include/nbl/builtin/hlsl/utility.hlsl

```diff
@@ -13,6 +13,8 @@ namespace nbl
 {
 namespace hlsl
 {
+template<typename T>
+const static bool always_true = true;
```

Comment:
btw for all these `decltype(expr)` its probably better to check `is_same_v<decltype(expr),EXPECTED_RETURN_TYPE>` than just `always_true<decltype(expr)>` so you don't match just anything and fall victim to implicit int-float or float-int on the return types

Reply:
https://godbolt.org/z/nroYjbaxf